app/javascript/components/textual_summary/table_list_view.jsx

```diff
@@ -28,21 +28,39 @@ const renderRow = (row, i, colOrder, rowLabel, onClick) => (
 export default function TableListView(props) {
```

Comment:
@David, I have a small suggestion, could you please check the below code - 

```
export default function TableListView(props) {
  const { headers, values, title } = props;

  // eslint-disable-next-line no-prototype-builtins
  const keyExists = (key) => headers.some((item) => item.hasOwnProperty(key));

  /** Function to generate rows for structured list. */
  const miqListRows = (list) => {
    const headerKeys = keyExists('key') ? headers.map((item) => (item.key)) : headers;
    return list.map((item) => (headerKeys.map((header) => item[header.toLowerCase()])));
  };

  return (
    <MiqStructuredList
      headers={keyExists('label') ? headers.map((item) => item.label) : headers}
      rows={miqListRows(values)}
      title={title}
      mode="table_list_view"
      onClick={() => props.onClick}
    />
  );
}

```

Reply:
@DavidResende0 Sorry, I thought the first iteration had a missing `else` branch and relied on the return to break out of the function.  Maybe I missed it but I thought it was like this where both returns were in the same function:

```js
if ...
  return
...
return
```

This example looks different to me, perhaps because the logic was moved into the `miqListRows` function, so there's only a single return.  I'm getting this from the general rule in ruby to only do returns in methods at the guard clause at the beginning of the method.  It's generally preferable for readability due to less complexity.

It's my bad if the original version had only 1 return in the function.

Would this mean `miqListComplexTable` gets removed?  I'm fine if the "complex" vs. "simple/default" table aren't named domain concepts but if they have differences in a few places, it might make sense to name them.

I'm fine with whatever works, has good readability, and limits complexity.

👍

Reply:
@jeffibm, I tested these changes and it breaks with translations since you're relying on the headers to be in English and not marked for translation.